apps/OpenSpace/ext/launcher/src/sgctedit/sgctedit.cpp

```diff
@@ -235,63 +376,101 @@ void SgctEdit::apply() {
         std::filesystem::create_directories(absPath(userCfgTempDir));
     }
     _saveTarget = userCfgTempDir + "/apply-without-saving.json";
-    _cluster = std::move(cluster);
     accept();
 }
 
-sgct::config::Cluster SgctEdit::generateConfiguration() const {
-    sgct::config::Cluster cluster;
-
-    sgct::config::Scene scene;
-    scene.orientation = _settingsWidget->orientation();
-    cluster.scene = std::move(scene);
+void SgctEdit::generateConfiguration() {
+    _cluster.scene = sgct::config::Scene();
+    _cluster.scene->orientation = _settingsWidget->orientation();
+    if (_cluster.nodes.size() == 0) {
+        _cluster.nodes.push_back(sgct::config::Node());
+    }
+    sgct::config::Node& node = _cluster.nodes.back();
 
-    cluster.masterAddress = "localhost";
+    generateConfigSetupVsync();
+    generateConfigUsers();
+    generateConfigAddresses(node);
+    generateConfigResizeWindowsAccordingToSelected(node);
+    generateConfigIndividualWindowSettings(node);
+}
 
+void SgctEdit::generateConfigSetupVsync() {
     if (_settingsWidget->vsync()) {
-        sgct::config::Settings::Display display;
-        display.swapInterval = 1;
-        
-        sgct::config::Settings settings;
-        settings.display = display;
-
-        cluster.settings = settings;
+        if (!_cluster.settings || !_cluster.settings.value().display ||
+            !_cluster.settings.value().display.value().swapInterval)
```

Comment:
-> `_cluster.settings->display->swapInterval`

also need to check that `display` actually has a value first or else it will crash